packages/php-wasm/universal/src/lib/base-php.ts

```diff
@@ -613,7 +613,51 @@ export abstract class BasePHP implements IsomorphicLocalPHP {
 	/** @inheritDoc */
 	@rethrowFileSystemError('Could not move "{path}"')
 	mv(fromPath: string, toPath: string) {
-		this[__private__dont__use].FS.rename(fromPath, toPath);
+		const mv = (fromPath: string, toPath: string) => {
+			const FS = this[__private__dont__use].FS;
+
+			const fromStat = FS.stat(fromPath);
+
+			let toStat;
+			try {
+				toStat = FS.stat(toPath);
+			} catch {
+				const parentPath = toPath.split('/').slice(0, -1).join('/');
+				try {
+					toStat = FS.stat(parentPath);
+				} catch {
+					toStat = { dev: NaN };
+				}
+			}
+
+			const fromMode = parseInt(
+				fromStat.mode.toString(8).substring(0, 1)
+			);
+			const fromIsDir = fromMode === 4;
+
+			if (fromStat.dev === toStat.dev) {
+				FS.rename(fromPath, toPath);
+			} else if (fromIsDir) {
+				try {
+					FS.mkdir(toPath);
+				} catch {
+					void 0;
+				}
+				const files = FS.readdir(fromPath);
+				files
+					.filter((f: string) => !['.', '..'].includes(f))
+					.forEach((f: string) =>
+						mv(fromPath + '/' + f, toPath + '/' + f)
```

Comment:
I suppose, theoretically, this line could use `this.mv()` instead of creating a new internal function `mv` on every call.

Reply:
@eliot-akira I thought the same thing, but this method is overridden in some use cases, and those overrides do not include calls to `super` so that breaks the recursion unless I create another function to recurse on.

Reply:
Oh I see, I didn't think about overrides, that makes sense!